src/Widgets/Block/DisplayTranslation.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace WPML\PB\Gutenberg\Widgets\Block;
+
+use WPML\LIB\WP\Hooks;
+use function WPML\Container\make;
+use function WPML\FP\spreadArgs;
+
+class DisplayTranslation implements \IWPML_Frontend_Action, \WPML\PB\Gutenberg\Integration {
+
+	const PRIORITY_BEFORE_REMOVE_BLOCK_MARKUP = 0;
+
+	public function add_hooks() {
+		Hooks::onFilter( 'widget_block_content', self::PRIORITY_BEFORE_REMOVE_BLOCK_MARKUP )
```

Comment:
This is a new filter add in WP 5.8.

My suggestion is to get the widget translations directly from the MO file and use it to replace strings in blocks. It does not add any DB query, and the whole logic will be simpler.

It's also consistent with the way we are currently translating strings in legacy widgets (on-the-fly).